R/utils-MakeTargetData.R

```diff
@@ -0,0 +1,29 @@
+#' Helper function to compile "long" group metadata
+#'
+#' @description
+#'
+#' Used to create percentage values on enrollment data, such as site activation and
+#' and participantment enrollment, and pasting together an appropriate n/N (xx.x%) value
+#' as well.
+#'
+#' @param data The input dataframe
+#' @param current_col Column that represents site count or participant count
+#' @param target_col Column that represents the target count for site/participants
+#' @param perc_val Name of column that will contain the numeric percentage value on enrollment
+#' @param perc_str_val Name of column that will contain the n/N (xx.x%) string
+#'
+#' @return A data frame containing two additional columns for the precentage value and associated string
+#' @export
+MakeTargetData <- function(data, current_col, target_col, perc_val, perc_str_val) {
+  if (!(current_col %in% names(data) & target_col %in% names(data))) {
+    cli::cli_abort("Check that both {current_col} and {target_col} are in data")
+  }
+
+  data <- data %>%
+    dplyr::mutate(
+      {{ perc_val }} := round(.data[[current_col]] * 100 / .data[[target_col]], 1),
```

Comment:
TIL dynamic lhs assignment 🤯

Reply:
I won't take all the credit, definitely a trick I learned from `admiral` dev team